### PR TITLE
use topkg for building charrua, instead of a handcrafted shell script

### PR DESCRIPTION
**Pull request by hannesm**
solves #34, please let me know what you think ;)

### CODE REVIEW

**Comment by yomimono** [comment]
I think it's great, thanks!

**Comment by haesbaert** [comment]
Heya, I was out and couldn't look at it. I like the idea ! If mindy is ok
I'm ok too.

**Comment by haesbaert** [comment]
Two small bits:

I would ask just to split into two commits, the topkg from the
whitespace/Result pruning in pcap.ml

Does it make sense to have build and build-test ? I mean if the tests dont
pass it shouldn't build. I'm afraid of getting lazy and running build
instead of build-test.

Also forcing opam to always run the tests when it builds, would make sure
that if something external to the pckage changes, we notice.

**Comment by hannesm** [comment]
On 30/01/2017 17:04, Christiano Haesbaert wrote:
> I would ask just to split into two commits, the topkg from the
> whitespace/Result pruning in pcap.ml

Sure!

> Does it make sense to have build and build-test ? I mean if the tests dont
> pass it shouldn't build. I'm afraid of getting lazy and running build
> instead of build-test.

Topkg has a build context (see
http://erratique.ch/software/topkg/doc/Topkg.Conf.html#VALbuild_context)
- and if you're locally doing a `ocaml pkg/pkg.ml build` in your cloned
git repository, it'll automatically choose the `Dev context and build
your tests.

`ocaml pkg/pkg.ml test` will then execute the test!

> Also forcing opam to always run the tests when it builds, would make sure
> that if something external to the pckage changes, we notice.

I don't think the time compiling and executing tests of released
packages during installation is well spent.  Changes in dependencies may
break a package - but there's likely soon infrastructure to build all
reverse dependencies (in test mode: building and executing them)
whenever a new release is done to the opam repository.

TL;DR: I like build and build-test.

**Comment by haesbaert** [comment]
Ah im cool if it picks the build-test on the dev environment, the other
point is not so critical.

With that Im ok.

**Comment by hannesm** [comment]
updated: split into 3 commits
